akka-http-tests/src/test/java/akka/http/javadsl/client/HttpAPIsTest.java

```diff
@@ -20,8 +22,17 @@
 
 import java.util.concurrent.CompletionStage;
 
+// compile only test, not runnable
 @SuppressWarnings("ConstantConditions")
-public class HttpAPIsTest extends JUnitRouteTest {
+public class HttpAPIsTest {
+
+  private ActorSystem system() {
+    throw new RuntimeException("compile time only");
+  }
+
+  private Materializer materializer() {
+    throw new RuntimeException("compile time only");
+  }
```

Comment:
Can we fix it the other way around? To see it in test listings (I'm panicing a bit about tests which I may expect to run but won't – we had this far too many times).

So we'd add a simple test here saying `@Test void compileTestOnlyCheckSources() {}`?